test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1

```diff
@@ -1,64 +1,78 @@
-        $EgJObject.Add("TestValue1", "123456")
-        $EgJObject.Add("TestValue2", "78910")
-        $EgJObject.Add("TestValue3", "99999")
-        $dict = @{}
-        $dict.Add('JObject', $EgJObject)
-        $dict.Add('StrObject', 'This is a string Object')
-        $properties = @{'DictObject' = $dict; 'RandomString' = 'A quick brown fox jumped over the lazy dog'}
-        $object = New-Object -TypeName psobject -Property $properties
+        $object = [pscustomobject]@{
+            HashTable = [hashtable]@{
+                JObject = [JObject]::FromObject( @{
+                        TestValue1 = 123456;
+                        TestValue2 = 78910;
+                        TestValue3 = 99999
+                    } );
+                StrObject = 'This is a string Object'
+            };
+            RandomString = 'A quick brown fox jumped over the lazy dog'
+        }
+
         $jsonFormat = ConvertTo-Json -InputObject $object
         $jsonFormat | Should -Match '"TestValue1": 123456'
         $jsonFormat | Should -Match '"TestValue2": 78910'
         $jsonFormat | Should -Match '"TestValue3": 99999'
     }
 
-	It "StopProcessing should succeed" {
-        $ps = [PowerShell]::Create()
-        $null = $ps.AddScript({
-            $obj = [PSCustomObject]@{P1 = ''; P2 = ''; P3 = ''; P4 = ''; P5 = ''; P6 = ''}
-            $obj.P1 = $obj.P2 = $obj.P3 = $obj.P4 = $obj.P5 = $obj.P6 = $obj
-            1..100 | Foreach-Object { $obj } | ConvertTo-Json -Depth 10 -Verbose
-            # the conversion is expected to take some time, this throw is in case it doesn't
-            throw "Should not have thrown exception"
-        })
-        $null = $ps.BeginInvoke()
-        # wait for verbose message from ConvertTo-Json to ensure cmdlet is processing
-        Wait-UntilTrue { $ps.Streams.Verbose.Count -gt 0 }
-        $null = $ps.BeginStop($null, $null)
-        # wait a bit to ensure state has changed, not using synchronous Stop() to avoid blocking Pester
-        Start-Sleep -Milliseconds 100
-        $ps.InvocationStateInfo.State | Should -BeExactly "Stopped"
+    It 'StopProcessing should succeed' {
+        $ps = [powershell]::Create()
+        [void]$ps.AddScript( {
+                $obj = [pscustomobject]@{P1 = ''; P2 = ''; P3 = ''; P4 = ''; P5 = ''; P6 = ''}
+                $obj.P1 = $obj.P2 = $obj.P3 = $obj.P4 = $obj.P5 = $obj.P6 = $obj
+                (1..100).ForEach{
+                    Write-Verbose -Message 'Ready' -Verbose
+                    ConvertTo-Json -InputObject $obj -Depth 10
+                }
+                throw 'ConvertTo-Json finished processing before it could be stopped.'
+            } )
+
+        [void]$ps.BeginInvoke()
+
+        # Wait until there is output in the verbose stream.
+        Wait-UntilTrue { $ps.Streams.Verbose.Count -gt 0 } -TimeoutInMilliseconds 1000 -IntervalInMilliseconds 10
+
+        # Wait to ensure ConvertTo-Json has started processing.
+        Start-Sleep -Milliseconds 200
```

Comment:
Using a sleep isn't sufficient since in CI, the container or VM could be slow so again the pipeline is stopped before ConvertTo-Json is actually started.  The original code used verbose output that comes out of ConvertTo-Json to detect it has started.